okgo/check/check.go

```diff
@@ -27,6 +27,69 @@ import (
 )
 
 func Run(projectParam okgo.ProjectParam, checkersToRun []okgo.CheckerType, pkgPaths []string, projectDir string, factory okgo.CheckerFactory, parallelism int, stdout io.Writer) error {
+	checkers, maxTypeLen, err := getCheckersToRun(projectParam, checkersToRun, factory)
+	if err != nil {
+		return err
+	}
+	// if there are fewer checkers than max parallelism, update parallelism to number of checkers
+	if len(checkers) < parallelism {
+		parallelism = len(checkers)
+	}
+
+	jobs := make(chan okgo.CheckerParam, len(checkers))
+	results := make(chan checkResult, len(checkers))
+
+	var checksWithFailures []string
+	pullResultsOff := func(toRun int) {
+		for i := 0; i < toRun; i++ {
+			checkResult := <-results
+			if checkResult.producedOutput {
+				checksWithFailures = append(checksWithFailures, string(checkResult.checkerType))
+			}
+		}
+	}
+
+	startASingleWorker := func() {
+		go singleCheckWorker(pkgPaths, projectDir, maxTypeLen, parallelism > 1, jobs, results, stdout)
+	}
+	// We always have 1 worker no matter what
+	startASingleWorker()
+
+	// First run all jobs that need to be ran one at a time
+	checkersWeCannotRunInParallel, checkersWeCanRunInParallel, err := partitionCheckerJobs(checkers)
+	if err != nil {
+		return err
+	}
+	// Start all the single ones
+	for _, checker := range checkersWeCannotRunInParallel {
+		jobs <- checker
+	}
+	// And then pull off those results
+	pullResultsOff(len(checkersWeCannotRunInParallel))
```

Comment:
```suggestion
	// Start all jobs that must be run serially. This enqueues all jobs in order and
	// because there is only one worker they will run one at a time.
	for _, checker := range checkersWeCannotRunInParallel {
		jobs <- checker
	}
	// Retrieve all results
	pullResultsOff(len(checkersWeCannotRunInParallel))
```

Reply:
Done